sig/password.rbs

```diff
@@ -0,0 +1,40 @@
+# Classes
+module Argon2
+  class Password
+    # Password constants
+    DEFAULT_T_COST: Integer
+    MIN_T_COST: Integer
+    MAX_T_COST: Integer
+    DEFAULT_M_COST: Integer
+    MIN_M_COST: Integer
+    MAX_M_COST: Integer
+
+    # Password attributes
+    attr_reader digest: String?
+    attr_reader checksum: String?
+    attr_reader salt: String?
+    attr_reader variant: String?
+    attr_reader version: Integer?
+    attr_reader t_cost: Integer?
+    attr_reader m_cost: Integer?
+    attr_reader p_cost: Integer?
```

Comment:
I'm not totally sure how exposing this works. I made them all `nil | Type` to be safe, but technically every single one of these will be set in the initializer without exception. If it's okay with them being nil during the initialization, we could make these all hard return values (`Type` only / remove the `?`)